src/features/codeLensProvider.ts

```diff
@@ -131,12 +133,24 @@ export default class OmniSharpCodeLensProvider extends AbstractProvider implemen
             }
 
             bucket.push(new vscode.CodeLens(
-                toRange(node.Location),
+                attributeRange,
                 { title: "run test", command: 'dotnet.test.run', arguments: [testFeature.Data, fileName, testFrameworkName] }));
 
             bucket.push(new vscode.CodeLens(
-                toRange(node.Location),
+                attributeRange,
                 { title: "debug test", command: 'dotnet.test.debug', arguments: [testFeature.Data, fileName, testFrameworkName] }));
         }
     }
+
+    private _attributeSpanToRange(document: vscode.TextDocument, node: protocol.Node): vscode.Range {
+        //If the node has some attributes then returns the range of the attributes, else returns the range from the actual node location
+        if (node.AttributeSpanStart) {
+            let line = document.positionAt(node.AttributeSpanStart).line;
+            let column = document.positionAt(node.AttributeSpanStart).character;
+            let endLine = document.positionAt(node.AttributeSpanEnd).line;
+            let endColumn = document.positionAt(node.AttributeSpanEnd).character;
+            return new Range(line, column, endLine, endColumn);
```

Comment:
I think there's a range constructor that takes 2 `Positions`s (instead of 2 l,c coordinates)